web_custom_modifier/models/ir_ui_view.py

```diff
@@ -0,0 +1,61 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import json
+from lxml import etree
+from typing import List
+from odoo import api, fields, models
+from odoo.osv.orm import transfer_node_to_modifiers, transfer_modifiers_to_node
+
+
+def _add_custom_modifier_to_view_tree(modifier: dict, tree: etree._Element):
+    """Add a custom modifier to the given view architecture.
+
+    :param modifier: the modifier to add to the view.
+    :param tree: the etree.Element of the view to extend.
+    """
+    xpath_expr = (
+        "//field[@name='{field_name}'] | //modifier[@for='{field_name}']"
+        .format(field_name=modifier['reference'])
+        if modifier['type_'] == 'field' else modifier['reference']
+    )
+
+    for node in tree.xpath(xpath_expr):
+        modifiers = {}
+        transfer_node_to_modifiers(node, modifiers)
+        modifiers[modifier['modifier']] = True
+        transfer_modifiers_to_node(modifiers, node)
+
+
+def _add_custom_modifiers_to_view_arch(modifiers: List[dict], arch: str) -> str:
+    """Add custom modifiers to the given view architecture.
+
+    :param arch: the architecture to extend.
+    :param modifiers: a list of custom modifiers to apply.
+    :return: the view architecture with custom modifiers.
+    """
+    if not modifiers:
+        return arch
+
+    tree = etree.fromstring(arch)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Using lxml.etree.fromstring to parse untrusted XML data is known to be vulnerable to XML attacks. Replace lxml.etree.fromstring with its defusedxml equivalent function.](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=4709570)